src/pudl/analysis/demand_mapping.py

```diff
@@ -1,294 +1,1006 @@
+            COUNTRY=lambda x: x.COUNTRY.astype(pd.StringDtype()),
+            NAICS_CODE=lambda x: x.NAICS_CODE.astype(pd.Int64Dtype()),
+            NAICS_DESC=lambda x: x.NAICS_DESC.astype(pd.StringDtype()),
+            SOURCE=lambda x: x.SOURCE.astype(pd.StringDtype()),
+            VAL_METHOD=lambda x: x.VAL_METHOD.astype(pd.StringDtype()),
+            WEBSITE=lambda x: x.WEBSITE.astype(pd.StringDtype()),
+            ABBRV=lambda x: x.ABBRV.astype(pd.StringDtype()),
+            YEAR=lambda x: x.YEAR.astype(pd.Int64Dtype()),
+            PEAK_LOAD=lambda x: x.PEAK_LOAD.astype(float),
+            PEAK_RANGE=lambda x: x.PEAK_RANGE.astype(float),
+            SHAPE_Length=lambda x: x.SHAPE_Length.astype(float),
+            SHAPE_Area=lambda x: x.SHAPE_Area.astype(float),
+        )
     )
-    new_df["area_derived"] = new_df["geometry"].area
+    return gdf
 
-    gdf_intermediate["intermediate_area_derived"] = gdf_intermediate[geom_intermediate].area
-    gdf_source["source_area_derived"] = gdf_source[geom_source].area
 
-    new_df = (new_df[[gdf_intermediate_col, gdf_source_col, "area_derived", "geometry"]]
-              .merge(gdf_intermediate[[gdf_intermediate_col, "intermediate_area_derived"]])
-              .merge(gdf_source[[gdf_source_col, "source_area_derived"]]))
+################################################################################
+# Demand allocation functions
+################################################################################
+def edit_id_set(row, new_id, id_set):
+    """
+    Editing ID sets by adding the new geometry ID if required.
+
+    This function edits original ID sets by adding the new geometry ID if
+    required. This function is called by another function
+    `complete_disjoint_geoms`
+    """
+    if row["geom_type"] == "geometry_new_int":
+        return frozenset(list(row[id_set]) + [new_id])
 
-    new_df["gdf_intermediate_intersection_fraction"] = new_df["area_derived"] / \
-        new_df["intermediate_area_derived"]
-    new_df["gdf_source_intersection_fraction"] = new_df["area_derived"] / \
-        new_df["source_area_derived"]
+    else:
+        return row[id_set]
 
-    # Deleting temporary columns defined in the original GeoDataFrames
-    del gdf_intermediate["intermediate_area_derived"]
-    del gdf_source["source_area_derived"]
 
-    new_df = new_df[[gdf_intermediate_col, gdf_source_col,
-                     "gdf_intermediate_intersection_fraction",
-                     "gdf_source_intersection_fraction"]]
+def polygonize_geom(geom):
+    """
+    Remove zero-area geometries from a geometry collection.
+
+    Strip zero-area geometries from a geometrical object.
+    (maybe a single geometry object or collection e.g. GeometryCollection)
+    This function is called by another function `complete_disjoint_geoms`.
+    """
+    if isinstance(geom, GeometryCollection):
 
-    return new_df
+        new_list = [a for a in list(geom) if type(a) in [
```

Comment:
Codacy found an issue: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5664147)